src/ameba/rule/lint/logical_without_paren.cr

```diff
@@ -0,0 +1,48 @@
+module Ameba::Rule::Lint
+  # A rule that disallows logical operators in method args without parenthesis.
+  #
+  # For example, this is considered invalid:
+  #
+  # ```
+  # if foo.includes? "bar" || foo.includes? "batz"
+  # end
+  # ```
+  #
+  # And need to be written as:
+  #
+  # ```
+  # if foo.includes?("bar") || foo.includes?("batz")
+  # end
+  # ```
+  #
+  # YAML configuration example:
+  #
+  # ```
+  # Lint/LogicalWithoutParenthesis:
+  #   Enabled: true
+  # ```
+  class LogicalWithoutParenthesis < Base
+    properties do
+      description "Disallows logical operators in method args without parenthesis"
+    end
+
+    MSG = "Logical operator in method args without parenthesis is not allowed"
```

Comment:
Rule name and description is not very descriptive nor accurate, let's change it to (following [RuboCop](https://github.com/rubocop/rubocop/blob/ac98713904b3e6f10605f00786e85bc3b3d0aa15/lib/rubocop/cop/lint/require_parentheses.rb) here):

```suggestion
  # ```
  # Lint/RequireParentheses:
  #   Enabled: true
  # ```
  class RequireParentheses < Base
    properties do
      description "Disallows method calls with no parentheses and a logical operator in the argument list"
    end

    MSG = "Use parentheses in the method call to avoid confusion about precedence"
```

This needs to be updated in other places too (incl. the rule filename).

Reply:
Yeah I had trouble coming up with a good name for this, I'll update. Thank you!